Library/Formula/docker-machine.rb

```diff
@@ -22,7 +22,9 @@ def install
     mkdir_p buildpath/"src/github.com/docker/"
     ln_sf buildpath, buildpath/"src/github.com/docker/machine"
 
+    ENV["PWD"] = buildpath/"src/github.com/docker/machine/"
```

Comment:
Actually it's not luck, $PWD is a shell build in and setting it *will* change your path.

Try this in your shell:

    export PWD=/

But I agree that `Dir.chdir` should not resolve the symlinked path to the absolute path.